lib/galaxy/tool_util/toolbox/base.py

```diff
@@ -898,6 +898,8 @@ def _load_tool_tag_set(
         except OSError as exc:
             exc_info = exc.errno != ENOENT
             log.error("Error reading tool configuration file from path '%s': %s", path, exc, exc_info=exc_info)
+        except AssertionError as exc:
```

Comment:
Maybe `InteractiveTool.__init__()` should raise a `ServerNotConfiguredForRequest` exception instead of using an assert (also the `ValueError()` there is unnecessary), so we can catch a more specific exception here?